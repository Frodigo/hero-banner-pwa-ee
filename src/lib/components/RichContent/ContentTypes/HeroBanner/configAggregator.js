import { getAdvanced } from "@magento/pagebuilder/lib/utils";

export default (node) => {
    const desktopImageElement = node.querySelector('[data-element="desktop_image"]');
    const mobileImageElement = node.querySelector('[data-element="mobile_image"]');
    const headingElement = node.querySelector('[data-element="heading"]');
    const buttonElement = node.querySelector('[data-element="button_url"]');

    return {
        desktopImage: desktopImageElement.getAttribute('src'),
        mobileImage: mobileImageElement.getAttribute('src'),
        heading: headingElement.innerText,
        buttonText: buttonElement.innerText,
        buttonUrl: buttonElement.getAttribute('href'),
        ...getAdvanced(node)
    };
};
