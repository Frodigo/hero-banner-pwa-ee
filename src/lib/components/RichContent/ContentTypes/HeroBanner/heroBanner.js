import React from 'react';
import { mergeClasses } from '@magento/venia-ui/lib/classify';
import defaultClasses from './heroBanner.css';
import { arrayOf, shape, string } from 'prop-types';

const HeroBanner = props => {
  const classes = mergeClasses(defaultClasses, props.classes);
  const {
    desktopImage,
    mobileImage,
    heading,
    buttonText,
    buttonUrl,
    textAlign,
    border,
    borderColor,
    borderWidth,
    marginTop,
    marginRight,
    marginBottom,
    marginLeft,
    paddingTop,
    paddingRight,
    paddingBottom,
    paddingLeft,
    cssClasses
  } = props;
  const rootStyles = {
    textAlign,
    border,
    borderColor,
    borderWidth,
    marginTop,
    marginRight,
    marginBottom,
    marginLeft,
    paddingTop,
    paddingRight,
    paddingBottom,
    paddingLeft
  };

  let image = desktopImage;
  if (mobileImage && window.matchMedia('(max-width: 768px)').matches) {
    image = mobileImage;
  }

  return (
    <div
      className={[classes.heroBlock, ...cssClasses].join(' ')}
      style={rootStyles}
    >
      <figure className={classes.heroBlockFigure}>
        <img className={classes.heroBlockImage} src={image} alt="figure" />
      </figure>
      <section className={classes.heroBlockTape}>
        <h2 className={classes.heroBlockHeading}>{heading}</h2>
        <a className={classes.heroBlockAction} href={buttonUrl}>
          {buttonText}
        </a>
      </section>
    </div>
  );
};

HeroBanner.propTypes = {
  classes: shape({
    heroBlock: string
  }),
  contentType: string,
  appearance: string,
  buttonText: string.isRequired,
  buttonUrl: string.isRequired,
  desktopImage: string.isRequired,
  heading: string.isRequired,
  textAlign: string,
  border: string,
  borderColor: string,
  borderWidth: string,
  marginTop: string,
  marginRight: string,
  marginBottom: string,
  marginLeft: string,
  paddingTop: string,
  paddingRight: string,
  paddingBottom: string,
  paddingLeft: string,
  cssClasses: arrayOf(string)
};

export default HeroBanner;
