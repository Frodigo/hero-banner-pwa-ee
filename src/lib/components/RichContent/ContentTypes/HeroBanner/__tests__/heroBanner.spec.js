import React from 'react';
import { render } from '@testing-library/react';
import HeroBanner from '../heroBanner';

const mockedProps = {
  appearance: 'default',
  border: '',
  borderColor: '',
  borderRadius: '',
  borderWidth: '',
  buttonText: 'Shop now!',
  buttonUrl: '#',
  children: [],
  contentType: 'hero_banner',
  cssClasses: ['hero-block'],
  desktopImage: 'http://placehold.it/200x200',
  heading: 'Seasonal styles up to 40%',
  marginBottom: '0px',
  marginLeft: '0px',
  marginRight: '0px',
  marginTop: '0px',
  mobileImage: 'http://placehold.it/200x200',
  paddingBottom: '0px',
  paddingLeft: '0px',
  paddingRight: '0px',
  paddingTop: '0px',
  textAlign: ''
};

test('Display hero banner on the page with desktop image', () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
      matches: true
    }))
  });

  const { asFragment } = render(<HeroBanner {...mockedProps} />);
  expect(asFragment).toMatchSnapshot();
});

test('Display hero banner on the page with mobile image', () => {
  Object.defineProperty(window, 'matchMedia', {
    writable: true,
    value: jest.fn().mockImplementation(() => ({
      matches: false
    }))
  });

  const { asFragment } = render(<HeroBanner {...mockedProps} />);
  expect(asFragment).toMatchSnapshot();
});
