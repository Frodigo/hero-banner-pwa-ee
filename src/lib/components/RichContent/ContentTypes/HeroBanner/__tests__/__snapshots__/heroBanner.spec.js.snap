// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Display hero banner on the page 1`] = `[Function]`;

exports[`Display hero banner on the page with desktop image 1`] = `[Function]`;

exports[`Display hero banner on the page with mobile image 1`] = `[Function]`;
